inc/sitemaps/class-sitemaps-cache.php

```diff
@@ -54,17 +54,59 @@ public function get_sitemap( $type, $page ) {
 	}
 
 	/**
+	 * Get the sitemap that is cached
+	 *
+	 * @param string $type Sitemap type.
+	 * @param int    $page Page number to retrieve.
+	 *
+	 * @return null|WPSEO_Sitemap_Cache_Data Null on no cache found otherwise object containing sitemap and meta data.
+	 */
+	public function get_sitemap_data( $type, $page ) {
+
+		$sitemap = $this->get_sitemap( $type, $page );
+
+		if ( empty( $sitemap ) ) {
+			return null;
+		}
+
+		// This should happend automatically, but it doesn't.
+		if ( 0 === strpos( $sitemap, 'C:24:"WPSEO_Sitemap_Cache_Data"' ) ) {
+			$sitemap = unserialize( $sitemap );
+		}
+
+		// What we expect it to be if it is set.
+		if ( $sitemap instanceof WPSEO_Sitemap_Cache_Data_Interface ) {
+			return $sitemap;
+		}
+
+		// Convert old type to new type, this will invalidate soon enough.
+		if ( is_string( $sitemap ) ) {
```

Comment:
Perhaps this can be better extracted to a separate function to keep this function smaller.